docs: add base URLs and authentication section to AI Bridge

matifali opened the pull request:
Add a concise Client Configuration section that explains how to configure
AI coding tools to use AI Bridge. This includes:

- Base URL configuration for OpenAI and Anthropic endpoints
- Authentication using Coder session tokens/API keys
- General compatibility notes without specific client details

This is extracted from #20340

matifali committed:
Update ai-bridge.md with additional template usage and refine compatibility notes

- Added a reference to using `data.coder_workspace_owner.me.session_token` in templates for generating Coder API keys.
- Reformatted the 'Compatibility Notes' section for better clarity, and revised some details about client-specific requirements.

matifali committed:
docs(ai-bridge): add client configuration section with base URLs

Add a concise Client Configuration section that explains how to configure
AI coding tools to use AI Bridge. This includes:

- Base URL configuration for OpenAI and Anthropic endpoints
- Authentication using Coder session tokens/API keys
- General compatibility notes without specific client details

This is a focused extraction from the broader aibridge-provider-docs branch,
containing only the essential setup information without client-specific guides.

matifali committed:
Merge branch 'main' into aibridge-setup

dannykopping reviewed (commented):
Looking good, a couple minor comments.

matifali commented on docs/ai-coder/ai-bridge.md:
````diff
@@ -81,6 +81,44 @@ Bridge is compatible with _[Google Vertex AI](https://cloud.google.com/vertex-ai
+coder tokens create
+```
+
+or in templates using the [`data.coder_workspace_owner.me.session_token`](https://registry.terraform.io/providers/coder/coder/latest/docs/data-sources/workspace_owner#session_token-1)
````
I was thinking to add the example in the follow up PR. WDYT?

matifali committed:
docs: address review feedback on AI Bridge client configuration

- Clarify that OPENAI_BASE_URL and ANTHROPIC_BASE_URL are commonly environment
  variables but configuration methods vary by client
- Reframe authentication section from operator perspective, referring to 'users'
  and 'their' keys rather than 'your' keys
- Make it clear operators don't distribute provider keys to users

matifali committed:
docs: clarify template authentication is for template admins

Update the session token configuration line to be explicitly for template
admins who are setting up authentication for their users, rather than
sounding like it's for end users themselves.

matifali commented on docs/ai-coder/ai-bridge.md:
```diff
@@ -81,6 +81,44 @@ Bridge is compatible with _[Google Vertex AI](https://cloud.google.com/vertex-ai
+
+### Authentication
+
+Instead of using provider-specific API keys (OpenAI/Anthropic keys), clients authenticate using your **Coder session token** or **API key**:
```
adjusted

deansheather commented on docs/ai-coder/ai-bridge.md:
```diff
@@ -81,6 +81,44 @@ Bridge is compatible with _[Google Vertex AI](https://cloud.google.com/vertex-ai
+
+#### Compatibility Notes
+
+Most AI coding assistants that support custom base URLs can work with AI Bridge. However, client-specific configuration requirements vary:
```
I think we should list compatibility notes for stuff we know about

matifali committed:
Elaborate on compatibility details for using AI Bridge with various AI coding assistants. Include specific examples such as removing the `/v1` suffix for URL formats and note clients like Cursor that proxy requests through their servers. Highlight instances where custom base URLs are not supported, as in Copilot CLI and Sourcegraph Amp.

dannykopping commented on docs/ai-coder/ai-bridge.md:
```diff
@@ -81,6 +81,44 @@ Bridge is compatible with _[Google Vertex AI](https://cloud.google.com/vertex-ai
+
+### Setting Base URLs
+
+Configure your AI client to point to your Coder deployment's AI Bridge endpoints. The exact configuration method varies by client—some use environment variables, others use configuration files or UI settings:
```
```suggestion
The exact configuration method varies by client — some use environment variables, others use configuration files or UI settings:
```

matifali committed:
Update docs/ai-coder/ai-bridge.md

Co-authored-by: Danny Kopping <danny@coder.com>

matifali committed:
Update docs/ai-coder/ai-bridge.md

Co-authored-by: Danny Kopping <danny@coder.com>
